packages/react-provider/src/ws.ts

```diff
@@ -224,7 +224,13 @@ export class WS {
 
   onReconnection(handler: { id: string; callback: () => void }): void {
     if (this.onReconnectionHandlers.find((h) => h.id === handler.id)) {
-      return;
+      this.onReconnectionHandlers = this.onReconnectionHandlers?.map((h) => {
+        if (h.id === handler.id) {
+          return handler;
+        }
+
+        return h;
+      });
```

Comment:
won't the find give out the h and handler matches anyway? do we even need the if?

Reply:
the if is if it isn't already in the array we need to push it

Reply:
coding is hard